api/base/filters.py

```diff
@@ -61,6 +63,78 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class MultiValueCharFilter(django_filters.BaseInFilter, django_filters.filters.CharFilter):
+
+    def filter(self, qs, value):
+        q = Q()
+        values = value or []
+        for value in values:
+            name = self.name
+            q |= Q(**{'{}__icontains'.format(name): value})
+        qs = qs.filter(q)
+
+        return qs
+
+
+class JSONAPIFilterSet(django_filters.FilterSet):
+
+    QUERY_PATTERN = re.compile(r'^filter\[(?P<fields>((?:,*\s*\w+)*))\](\[(?P<op>\w+)\])?$')
+    FILTER_FIELDS = re.compile(r'(?:,*\s*(\w+)+)')
+
+    or_fields = {}
+
+    def __init__(self, data=None, *args, **kwargs):
+        self.or_fields = {}
+        field_names = []
+        if data:
+            new_data = {}
+            for key, value in data.iteritems():
+                match = self.QUERY_PATTERN.match(key)
+                if match:
+                    match_dict = match.groupdict()
+                    fields = match_dict['fields']
+                    field_names = re.findall(self.FILTER_FIELDS, fields.strip())
+                    values = value.split(',')
+                    if len(field_names) > 1:
+                        self.or_fields[frozenset(field_names)] = values
+                    for field in field_names:
+                        new_data.update({field: value})
+            data = new_data
+        super(JSONAPIFilterSet, self).__init__(data=data, *args, **kwargs)
+
+        for field in field_names:
```

Comment:
I at first tried to do this check inside of the loop itself, but the only way I could access the fields that were supposed to be there was to access the form fields -- which is inaccessible until `super(JSONAPIFilterSet, self).__init__(...)` is called -- since `self.form.fields` isn't set until `self.filters` is set, which happens in the `super`'s init. 

The check is necessary here (or somewhere, at least), as the check in `filter_groups` only checks for invalid fields that are included in a group that will be part of an OR query (eg `?filter[wrong,field]`) and not filters that will go through normal filtering later (eg `?filter[wrong_field]=whateves`)

There might be a way to instead deal with this using [django-filter's strict validation](http://django-filter.readthedocs.io/en/develop/ref/filterset.html#strict) -- I tried for quite a long time to get this to work, and concluded that it was checking for an invalid filter **value** rather than an invalid filter itself -- which gets overlooked as far as I can tell by django-filter by default.